docs/softwaredevelopment.md

```diff
@@ -1 +1,124 @@
+
+#### Go
+Go is an appropriate language for systems programming, like proxying, routing, and transforming HTTP requests. However you should only write these sorts of components if there is no alternative maintained open source tool available.
+
+Go has a feature complete standard library, good concurrency primitives and is a memory safe language. These features make it a good choice for backend application which aggregate or transform APIs, or have performance requirements.
+
+## Style Guides
+
+### Code style guides
+Developers read code much more often than they write it. These guidelines are intended to improve the readability of code and make it consistent across DDS projects.
+
+A style guide is about consistency. Consistency with this style guide is important. Consistency within a project is more important. Consistency within one module or function is most important.
+
+But most importantly: know when it’s ok to be inconsistent. Sometimes the style guide just does not apply. When in doubt, use your best judgement. Look at other examples and decide what looks best. And do not hesitate to ask if you’re unsure.
+
+Some good reasons to ignore a particular guideline include:
+
+- when applying the guideline would make the code less readable, even for someone who is used to reading code that follows this style guide
+- to be consistent with surrounding code that also breaks it (maybe for historic reasons) – although this is also an opportunity to clean up the existing code
+ - when the code in question is older than the introduction of the guideline and there is no reason to modify that code
+- when the code needs to remain compatible with older versions that do not support the feature recommended by the style guide
+
+## How to manage third party software dependencies
+When you develop and operate a service, it’s important to keep any third party dependencies you use up-to-date. By doing this, you can avoid potential security vulnerabilities.
+
+Any automated tools you use to manage third party dependencies should be compatible with DDS supported programming languages. The tools you use should neither slow down your development process nor disclose potential security vulnerabilities to the public.
+
+You can read more about [managing software dependencies in the Service Manual](https://www.gov.uk/service-manual/technology/managing-software-dependencies), where you will find a list of common dependency management tools.
+
+### Update dependencies frequently
+Update your dependencies frequently rather than in ‘big bang’ batches. This works well with continuous delivery principles and makes sure the changes introduced are small and can be automatically tested.
+
+There are tools which scan GitHub repositories and raise PRs when they find dependency updates. 
+
+- [Dependabot](https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/) 
+
```

Comment:
Might want to remove this bullet (since there are no others) and replace with "Dependabot [link] is enabled by default on all new and existing repositories."